### PR TITLE
Add Email Verification

### PR DESCRIPTION
**Pull request by Deichor**
### Summary

I have implemented a feature to enforce email verification and allow specifying a column for this purpose.  

### Changes Introduced
- **Two new settings have been added:**
  - `kickNotVerified`: Prevents users with unverified emails from logging into the server.
  - `mySQLVerifiedField`: Specifies the name of the field in the database to be checked for email verification.

- **Additional Updates:**
  - Ensured compatibility with various database systems and conducted thorough testing. No issues were observed during the tests.
  - Did not add functionality to kick users immediately after registering, as this feature already exists. The new settings are designed to be combined for flexibility.
  - Updated the repository URL and version of **MultiverseCore**, as its repository has changed.
  - Updated the Log4J version to the latest one.

### Notes
- These changes aim to improve flexibility and ensure better integration with email verification systems.  
- The updates for MultiverseCore and Log4J address issues with outdated dependencies.

### CODE REVIEW

File: src/main/java/fr/xephi/authme/listener/PlayerListener.java [hunk 1]
```diff
@@ -150,8 +150,9 @@ public void onAsyncPlayerPreLoginEventHighest(AsyncPlayerPreLoginEvent event) {
         // Slow, blocking checks
         try {
             final PlayerAuth auth = dataSource.getAuth(name);
-            final boolean isAuthAvailable = auth != null;
+            final boolean isAuthAvailable= auth != null;
```
Comment (rautamiekka) [inline src/main/java/fr/xephi/authme/listener/PlayerListener.java:153]:
```suggestion
            final boolean isAuthAvailable = auth != null;
```

**Review by rautamiekka — CHANGES_REQUESTED** [review]

File: src/main/java/fr/xephi/authme/message/MessageKey.java
Comment (HaHaWTH) [inline src/main/java/fr/xephi/authme/message/MessageKey.java:1]: Should be changed to `email.unverified_kick`

File: src/main/java/fr/xephi/authme/settings/properties/RestrictionSettings.java [hunk 2]
```diff
@@ -100,6 +100,10 @@ public final class RestrictionSettings implements SettingsHolder {
     public static final Property<Boolean> KICK_NON_REGISTERED =
         newProperty("settings.restrictions.kickNonRegistered", false);
 
+    @Comment("Should unregistered players be kicked immediately?")
```
Comment (HaHaWTH) [inline src/main/java/fr/xephi/authme/settings/properties/RestrictionSettings.java:103]: Here should be `unverified`, just like changes in docs/config.md

File: src/main/resources/messages/messages_en.yml
Comment (HaHaWTH) [inline src/main/resources/messages/messages_en.yml:1]: Can you add this message key to other language files? Translator usage is fine, or I can do it later🙂

**Review by HaHaWTH — CHANGES_REQUESTED** [review]

File: src/main/resources/messages/messages_fi.yml
Comment (HaHaWTH) [inline src/main/resources/messages/messages_fi.yml:1]: Here should be `unverified_kick`, just like other lang files

File: src/main/resources/messages/messages_eu.yml
Comment (HaHaWTH) [inline src/main/resources/messages/messages_eu.yml:1]: Same as messages_fi.yml

**Review by HaHaWTH — APPROVED** [review]

**Comment by HaHaWTH** [comment]
Merged, thanks!

**Pull request merged**